metrics/ci_integrations/docs/diagrams/commands_code_sharing.puml

```diff
@@ -0,0 +1,135 @@
+@startuml commands_code_sharing
+
+package cli {
+    package command {
+        abstract class CiIntegrationCommand {
+            + dynamic getArgumentValue(String argumentName)
+        }
+
+        class ValidateCommand {
+            + rawIntegrationConfigFactory : RawIntegrationConfigFactory
+            + configuredPartiesFactory : ConfiguredPartiesFactory
+
+            + Future<void> run()
+        }
+    }
+
+    package parties as cli_parties {
+        abstract class Parties<T extends IntegrationParty> {
+            + parties : List<T>
+
+            + T getParty(Map<String, dynamic> config)
+        }
+
+        class SupportedSourceParties {}
+
+        class SupportedDestinationParties {}
+
+        class SupportedIntegrationParties {}
```

Comment:
```suggestion
        class SupportedIntegrationParties {
        + sourceParties : Parties<SourceParty> 
        + destinationParties : Parties<DestinatinoParty>
        }
```
Would it be more clear? Also, it seems like this class uses the `SourceParty` and `DestinatinoParty` abstract classes.